alica_engine/src/engine/AlicaContext.cpp

```diff
@@ -78,8 +77,13 @@ bool AlicaContext::isValid() const
 void AlicaContext::stepEngine()
 {
     _engine->stepNotify();
+    constexpr const auto timeout = std::chrono::seconds(2);
```

Comment:
can use some factor of ALICA_LOOP_TIME_ESTIMATE instead of a magic number. Should also use the alica clock i.e. getAlicaClock